src/org/opensolaris/opengrok/configuration/Project.java

```diff
@@ -215,16 +215,15 @@ public static Project getProject(File file) {
      * @return project that fits the name
      */
     public static Project getByName(String name) {
-        Project ret = null;
         RuntimeEnvironment env = RuntimeEnvironment.getInstance();
         if (env.hasProjects()) {
             for (Project proj : env.getProjects()) {
-                if (name.indexOf(proj.getName()) == 0) {
-                    ret = proj;
+                if (name.equals(proj.getName())) {
```

Comment:
You probably mean `HashMap<String,Project>` ? The `getByName()` method needs to search according to `name` and return `Project` object.